### PR TITLE
fix(Consumer): remove auto creation of failure topics

### PR DESCRIPTION
**Pull request by srosolio**
https://jira.yotpo.com/browse/YO-14836

### CODE REVIEW

File: lib/yotpo_kafka/base_consumer.rb [hunk 1]
```diff
@@ -49,31 +50,11 @@ def subscribe_to_topics
...
-                 partitions_num: @partitions_num,
-                 replication_factor: @replication_factor)
-        @kafka.create_topic(failure_topic,
-                            num_partitions: @partitions_num.to_i,
-                            replication_factor: @replication_factor.to_i)
-      rescue Kafka::TopicAlreadyExists
-        nil
+      if @listen_to_failures
+        @consumer.subscribe(@failures_topic, start_from_beginning: @start_from_beginning)
+        log_info('Consumer subscribes to topic: ' + @failures_topic, broker_url: @seed_brokers)
```
Comment (omamiya) [inline lib/yotpo_kafka/base_consumer.rb:56]: `Consumer subscribes to topic` -> `Consumer subscribes to failures topic`

Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb:56]: + in `start_consumer` function, we catch the exception. lets add a field: `failures_topic`.

File: lib/yotpo_kafka/base_consumer.rb [hunk 2]
```diff
@@ -11,11 +11,12 @@ def initialize(params = {})
       @seed_brokers = params[:broker_url] || ENV['BROKER_URL'] || '127.0.0.1:9092'
       @kafka = Kafka.new(@seed_brokers)
       @seconds_between_retries = params[:seconds_between_retries] || 0
-      @listen_to_failures = params[:listen_to_failures].nil? ? true : params[:listen_to_failures]
+      @listen_to_failures = params[:listen_to_failures].nil? ? false : params[:listen_to_failures]
```
Comment (omamiya) [inline lib/yotpo_kafka/base_consumer.rb:14]: why?

Comment (srosolio) [reply lib/yotpo_kafka/base_consumer.rb:14]: absence should mean false, specially if another parameter is needed here(@failures_topic)

what would it mean to fail on not specifying listen_to_failures and not specifying failure_topic?

Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb:14]: so lets use only 1 parameter

**Review by omamiya — APPROVED** [review]

**Pull request merged**